frontend-react/src/hooks/api/messages/UseTestMessageResult/UseTestMessageResult.ts

```diff
@@ -0,0 +1,75 @@
+import { useQuery } from "@tanstack/react-query";
+import { useCallback, useState } from "react";
+import { useParams } from "react-router";
+import { reportsEndpoints, RSMessageResult } from "../../../../config/endpoints/reports";
+import useSessionContext from "../../../../contexts/Session/useSessionContext";
+import useAdminSafeOrganizationName, {
+    Organizations,
+} from "../../../UseAdminSafeOrganizationName/UseAdminSafeOrganizationName";
+
+const { testResult } = reportsEndpoints;
+
+/**
+ * Custom hook to fetch validation for a single test report.
+ *
+ * @returns {object} The hook returns the following:
+ * @property {RSMessageResult | []} data - The fetched test message results (empty array if none).
+ * @property {function} setRequestBody - A setter for the request body used by the query.
+ * @property {boolean} isLoading - `true` while the query is fetching.
+ * @property {boolean} isError - `true` if the query encountered an error.
+ * @property {Error | null} error - The error object if any error occurred.
+ * @property {"loading" | "error" | "success" | "idle"} status - The status of the query.
+ * @property {boolean} isDisabled - Indicates whether the feature is disabled (for non-admin users).
+ */
+const useTestMessageResult = () => {
+    const { activeMembership, authorizedFetch } = useSessionContext();
+    const { receivername } = useParams();
+    const parsedName = activeMembership?.parsedName;
+    const isAdmin = Boolean(parsedName) && parsedName === Organizations.PRIMEADMINS;
+    const adminSafeOrgName = useAdminSafeOrganizationName(parsedName);
+
+    const [requestBody, setRequestBody] = useState<string | null>(null);
+
+    const fetchData = useCallback(async () => {
+        try {
+            // Attempt the fetch
+            const result = await authorizedFetch<RSMessageResult>(
+                {
+                    params: {
+                        receiverName: receivername,
+                        organizationName: adminSafeOrgName,
+                    },
+                    data: requestBody,
+                },
+                testResult,
+            );
+
+            return result;
+        } catch (err) {
+            // Ensure we're rejecting with an actual Error object
+            if (err instanceof Error) {
+                return Promise.reject(err);
+            } else {
+                return Promise.reject(new Error(String(err)));
+            }
+        }
+    }, [authorizedFetch, receivername, adminSafeOrgName, requestBody]);
+
+    // Use 'enabled' to conditionally run the query whenever `requestBody` changes
+    // and the user is an admin. If requestBody is empty or user isn't admin, no fetch is made.
+    const { data, isLoading, status } = useQuery<RSMessageResult, Error>({
+        queryKey: [testResult.queryKey, activeMembership, receivername, adminSafeOrgName, requestBody],
+        queryFn: fetchData,
+        enabled: isAdmin && Boolean(requestBody),
+    });
+
+    return {
```

Comment:
Please return everything as-is from the hook, adding on anything else we want.

Reply:
This is actually an option within the `useQuery` hook that allows us to manually refetch the data: https://tanstack.com/query/v4/docs/framework/react/reference/useQuery.

Reply:
Our hooks need to always return everything the useQuery/useSuspenseQuery/useMutation hooks return and only add additional properties